src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -68,6 +70,10 @@ public ExamResource(UserService userService, ExamService examService, ExamAccess
             throw new BadRequestAlertException("A new exam cannot already have an ID", ENTITY_NAME, "idexists");
         }
 
+        if (exam.getCourse() == null) {
```

Comment:
Should exam here actually have a course? I mean the courseid is already defined through the path variable, isn't it?
If the issue would be that the provided entity is wrong you should consider HTTP 422 Unprocessable Entity
If considering, adapt tests as well.

Reply:
Yeah it should have a course, otherwise we have to set the course manually before we do `examService.save(exam)`. Still, it's good that you commented here. We should check if the course of the delivered exam matches the `courseId` in the route as we check if the user is at least instructor in the course with `courseId`. I will adapt that.

Reply:
Addition: I think I will stick with `conflict` as HTTP status here as we do not interpret the status in the client and I don't want to introduce too many new status to the project.

Reply:
To be honest I think that's bad design. There are standards for a reason and using non fitting statuses can make it hard for e.g. new developers as well.
However I don't what to fixiate on this, if it's not handled clientside it's not a breaking issue and you can resolve these comments if you like.